### PR TITLE
Add load_assignment field in Cluster

### PR DESCRIPTION
**Pull request by dio**
Add `load_assignment` field in Cluster

This patch introduces `load_assigment` field in CDS' Cluster. This is an API change only. 
This is part of effort on breaking https://github.com/envoyproxy/envoy/pull/3261 into multiple PRs.

*Risk Level*: 
- Low, since it is hidden.

*Testing*: 
- Build api and envoy-static without error

*Docs Changes*:
- Add load_assignment in Cluster of cds.proto.

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

File: api/envoy/api/v2/cds.proto [hunk 1]
```diff
@@ -157,6 +158,21 @@ message Cluster {
...
+  // This field supersedes :ref:`hosts<envoy_api_field_Cluster.hosts>` field.
+  //
+  // .. attention::
+  //
+  //   Setting this allows CDS static/DNS assignments to contain embedded EDS equivalent
+  //   :ref:`endpoint assignments<envoy_api_msg_ClusterLoadAssignment>`.
+  //   Setting this overrides :ref:`hosts<envoy_api_field_Cluster.hosts>` values.
+  //
+  // [#not-implemented-hide:]
+  ClusterLoadAssignment load_assignment = 33;
```
Comment (zuercher) [inline api/envoy/api/v2/cds.proto]: I think the discussion at https://github.com/envoyproxy/envoy/pull/3261/files#r188243657 still applies.

Comment (dio) [reply api/envoy/api/v2/cds.proto]: @zuercher thanks, will update it.

Comment (dio) [reply api/envoy/api/v2/cds.proto]: Updated in 0001298.

File: api/envoy/api/v2/cds.proto [hunk 2]
```diff
@@ -157,6 +158,21 @@ message Cluster {
...
+  // :ref:`STATIC<envoy_api_enum_value_Cluster.DiscoveryType.STATIC>`,
+  // :ref:`STRICT_DNS<envoy_api_enum_value_Cluster.DiscoveryType.STRICT_DNS>`
+  // or :ref:`LOGICAL_DNS<envoy_api_enum_value_Cluster.DiscoveryType.LOGICAL_DNS>` clusters.
+  // This field supersedes :ref:`hosts<envoy_api_field_Cluster.hosts>` field.
+  //
+  // .. attention::
+  //
+  //   Setting this allows CDS static/DNS assignments to contain embedded EDS equivalent
+  //   :ref:`endpoint assignments<envoy_api_msg_ClusterLoadAssignment>`.
+  //   Setting this overrides :ref:`hosts<envoy_api_field_Cluster.hosts>` values.
```
Comment (mattklein123) [inline api/envoy/api/v2/cds.proto]: @dio can you make a note to deprecate the hosts field and add to deprecated.md when you do the PR implementing this?

File: api/envoy/api/v2/cds.proto [hunk 3]
```diff
@@ -157,6 +158,21 @@ message Cluster {
...
 
+  // Setting this is required for specifying members of
+  // :ref:`STATIC<envoy_api_enum_value_Cluster.DiscoveryType.STATIC>`,
+  // :ref:`STRICT_DNS<envoy_api_enum_value_Cluster.DiscoveryType.STRICT_DNS>`
+  // or :ref:`LOGICAL_DNS<envoy_api_enum_value_Cluster.DiscoveryType.LOGICAL_DNS>` clusters.
+  // This field supersedes :ref:`hosts<envoy_api_field_Cluster.hosts>` field.
+  //
+  // .. attention::
+  //
+  //   Setting this allows CDS static/DNS assignments to contain embedded EDS equivalent
```
Comment (mattklein123) [inline api/envoy/api/v2/cds.proto]: "CDS static/DNS" is a bit confusing to read. Can you rephrase? Do you mean "Setting this allows non-EDS cluster types to contain..." ?

File: api/envoy/api/v2/cds.proto (hunk 2 above)
Comment (dio) [reply api/envoy/api/v2/cds.proto]: Sure, 2ab4f13. Thanks for reminding me.
